### PR TITLE
Hack pulp namespace support into galaxy_ng

### PR DESCRIPTION
**Pull request by newswangerd**
This adds "support" for pulp ansible namespace metadata by making it so that whenever galaxy_ng namespaces are updated, corresponding pulp ansible namespace metadata is added to any repo that contains a collection under the namespace.

Requires: https://github.com/pulp/pulp_ansible/pull/1395

Todo:
- [x] tests
- [x] make pulp ansible sync will work with the galaxy_ng namespace API.

### CODE REVIEW

File: galaxy_ng/app/migrations/0037_namespace_sync.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        if create_metadata:
+            metadata["name"] = old_ns.name
+            AnsibleNamespaceMetadata.objects.create(
+                **{
+                    **metadata,
+                    "metadata_sha256": calculate_metadata_sha256(metadata),
+                    "namespace": new_ns,
+                    "pulp_type": "ansible.namespace",
+                }
+            )
```
Comment (gerrod3) [inline galaxy_ng/app/migrations/0037_namespace_sync.py:73]:
```suggestion
            AnsibleNamespaceMetadata.objects.create(namespace=new_ns, **metadata)
```
`pulp_type` is auto added on save and `metadata_sha256` is also auto-calculated before the save. You can remove `calculate_metadata_sha256`

Comment (newswangerd) [reply galaxy_ng/app/migrations/0037_namespace_sync.py]: I don't think those hooks are called during migrations, but I can try.

File: galaxy_ng/app/access_control/statements/pulp.py [hunk 2]
```diff
@@ -195,6 +195,20 @@
...
+        "statements": [
+            {
+                "action": ["list", "retrieve"],
+                "principal": "authenticated",
+                "effect": "allow",
+            },
+            {
+                "action": "avatar",
+                "principal": "*",
+                "effect": "allow",
```
Comment (awcrosby) [inline galaxy_ng/app/access_control/statements/pulp.py:208]: is there a reason the `principal` is different on these 2 read namespace actions?

Comment (newswangerd) [reply galaxy_ng/app/access_control/statements/pulp.py:208]: yes. I wanted the avatar to be accessible to unauthenticated clients.

File: galaxy_ng/app/migrations/0038_namespace_sync.py [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    for each namespace:
+    - create AnsibleNamespaceMetadata object
+        - calculate the metadata sha
+    - add namespace permissions to pulp namespace obj
+        - or maybe not? should we just enforce the rbac on our
+          namespaces?
+
+    Add the new namespace to the following repos:
+        - published
+        - should these be added to the rest?
```
[the remark this answers is not in the document]
Comment (newswangerd) [reply galaxy_ng/app/migrations/0038_namespace_sync.py:30]: not having a namespace object in the repo is fine, it doesn't break anything. I figured that for PAH, the namespace logos will get added when a sync happens. Aside from that, the only repo with content in it that people care about is published.

**Review by awcrosby — APPROVED** [review]
After tests pass, LGTM

**Pull request merged**